apps/server-monolith/app/models/user.rb

```diff
@@ -0,0 +1,145 @@
+# frozen_string_literal: true
+
+# == Schema Information
+#
+# Table name: pods
+#
+#  id                                                                         :bigint           not null, primary key
+#  bio("Bio" means Biography in social media.)                                :string(140)
+#  deleted_at                                                                 :datetime
+#  display_name                                                               :string           not null
+#  external_avatar_url                                                        :string
+#  last_block_ids                                                             :json             not null
+#  last_pod_username                                                          :string
+#  suspended_at(the date when the user was suspended)                         :datetime
+#  suspended_reason(enumeration value for the reason for the user suspension) :integer          default(0)
+#  type                                                                       :enum             not null
+#  username                                                                   :string           not null
+#  created_at                                                                 :datetime         not null
+#  updated_at                                                                 :datetime         not null
+#
+# Indexes
+#
+#  index_pods_on_deleted_at           (deleted_at)
+#  index_pods_on_lower_username_text  (lower((username)::text)) UNIQUE
+#  index_pods_on_type                 (type)
+#
+class User < Pod
```

Comment:
User model doesn't care about the details of how authentication works.

`User::Authentication` should be transparent to the user.  Implement the `bind_or_create_user` method in User::Authentication